cfgov/v1/models/snippets.py

```diff
@@ -39,6 +49,93 @@ def get_by_title_slug(self, title, slug):
             hash=hashlib.md5(title + ';;' + slug).hexdigest())
 
 
+class ResourceTag(TaggedItemBase):
+    content_object = ParentalKey('v1.Resource', related_name='tagged_items')
+
+
+class TaggableSnippetManager(models.Manager):
+    def filter_by_tags(self, tags):
+        snippets = self.all()
+        for tag in tags or []:
+            snippets = snippets.filter(tags__name=tag)
+
+        return snippets
+
+
+@register_snippet
+class Resource(ClusterableModel):
+    title = models.CharField(max_length=255)
+    desc = RichTextField(verbose_name='Description', blank=True)
+
+    thumbnail = models.ForeignKey(
+        'v1.CFGOVImage',
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
```

Comment:
On this particular line, it wouldn't have a very adverse affect, because the thumbnail is a bonus thing that isn't actually rendered right now, and may never be.

On the other hand, If a Document selected for the `related_file` or `alternate_file` fields below was deleted, that would potentially be a more serious breakage of the snippet in question.

Another question about protecting: What is the messaging like when someone tries to delete that Image or Document? Is it smart enough to say something specific like "Hey now, snippet XYZ uses that Document. I can't let you delete that, Dave." If not, it could be maddening wanting to delete a file and not knowing where to go to remove it and un-protect it.